virtual_rainforest/models/abiotic_simple/simple_regression.py

```diff
@@ -0,0 +1,677 @@
+        "vapor_pressure_deficit_ref"
+    )
+
+
+def interpolate_soil_temperature(
+    layer_heights: DataArray,
+    layer_roles: List,
+    surface_temperature: DataArray,
+    mean_annual_temperature: DataArray,
+    lower_bound: float = -10,
+    upper_bound: float = 50,
+) -> DataArray:
+    """Interpolate soil temperature using logarithmic function.
+
+    Args:
+        layer_heights: vertical layer heights, [m]
+        layer_roles: list of layer roles (from top to bottom: above, canopy, subcanopy,
+            surface, soil)
+        surface_temperature: surface temperature, [C]
+        mean_annual_temperature: mean annual temperature, [C]
+        lower bound: minimum allowed values
+        upper_bound: maximum allowed value
+
+    Returns:
+        soil temperature profile, [C]
+    """
+
+    # select surface layer (atmosphere)
+    surface_layer = layer_heights[layer_heights.coords["layer_roles"] == "surface"]
+
+    # create array of interpolation heights including surface layer and soil layers
+    interpolation_heights = xr.concat(
+        [
+            surface_layer,
+            layer_heights[layer_heights.coords["layer_roles"] == "soil"] * -1
+            + surface_layer.values,
+        ],
+        dim="layers",
+    )
+
+    # Calculate per cell slope and intercept for logarithmic soil temperature profile
+    slope = (surface_temperature - mean_annual_temperature) / (
+        np.log(interpolation_heights.isel(layers=0))
+        - np.log(interpolation_heights.isel(layers=-1))
+    )
+    intercept = surface_temperature - slope * np.log(
+        interpolation_heights.isel(layers=0)
+    )
+
+    # Calculate the values within cells by layer
+    layer_values = np.log(interpolation_heights) * slope + intercept
+
+    # set upper and lower bounds
+    layer_values_low = np.where(layer_values < lower_bound, lower_bound, layer_values)
+    layer_values_bound = DataArray(
+        np.where(layer_values_low > upper_bound, upper_bound, layer_values_low),
+        dims=["layers", "cell_id"],
+        coords={
+            "layers": np.arange(0, len(interpolation_heights)),
+            "layer_roles": ("layers", layer_roles[0 : len(interpolation_heights)]),
+            "cell_id": mean_annual_temperature.cell_id,
+        },
+    )
+
+    return xr.concat(
+        [
+            DataArray(
+                np.full(
+                    (
+                        len(layer_roles) - len(interpolation_heights),
+                        len(mean_annual_temperature.coords["cell_id"]),
+                    ),
+                    np.nan,
+                ),
+                dims=["layers", "cell_id"],
+                coords={
+                    "layers": np.arange(len(interpolation_heights), len(layer_roles)),
+                    "layer_roles": (
+                        "layers",
+                        layer_roles[len(interpolation_heights) : len(layer_roles)],
+                    ),
```

Comment:
As mentioned above, unless I've misinterpreted the order of the layers, this should be the other way around.